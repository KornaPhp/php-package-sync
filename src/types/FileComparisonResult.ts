/* eslint-disable no-unused-vars */

export enum ComparisonKind {
    PACKAGE_NOT_USED = 'missing_pkg',
    PACKAGE_VERSION_MISMATCH = 'pkg_version',
    PACKAGE_SCRIPT_NOT_FOUND = 'pkg_script',

    FILE_NOT_FOUND = 'missing_file',
    DIRECTORY_NOT_FOUND = 'missing_dir',

    FILE_DOES_NOT_MATCH = 'not_exact',
    FILE_NOT_SIMILAR_ENOUGH = 'not_similar',

    FILE_NOT_IN_SKELETON = 'extra_file',
    DIRECTORY_NOT_IN_SKELETON = 'extra_dir',

    ALLOWED_SIZE_DIFFERENCE_EXCEEDED = 'size_diff',
}

export interface FileComparisonResult {
    kind: ComparisonKind;
    score: string | number;
    name: string;
    context?: any;
    skeletonPath: string;
    repositoryPath: string;
}
