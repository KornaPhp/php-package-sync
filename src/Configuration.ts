import { readFileSync } from 'fs';
import { ComparisonKind } from './types/FileComparisonResult';
import { PackageIssue } from './issues/PackageIssue';
import { ScoreRequirements } from './types/ScoreRequirements';

const yaml = require('js-yaml');

export interface ConfigurationRecord {
    fixers: {
        disabled?: string[];
        OptionalPackages: string[];
    };

    scoreRequirements: ScoreRequirements;
    ignoreNames: Array<string>;
    skipComparisons: Array<string>;
    templatesPath: string;
    packagesPath: string;
    templates: {
        vendor: string;
        names: string[];
    };
    issues: {
        ignored: {
            [ComparisonKind.DIRECTORY_NOT_FOUND]?: string[];
            [ComparisonKind.DIRECTORY_NOT_IN_SKELETON]?: string[];
            [ComparisonKind.FILE_DOES_NOT_MATCH]?: string[];
            [ComparisonKind.FILE_NOT_IN_SKELETON]?: string[];
            [ComparisonKind.FILE_NOT_SIMILAR_ENOUGH]?: string[];
            [ComparisonKind.PACKAGE_NOT_USED]?: string[];
            [ComparisonKind.PACKAGE_SCRIPT_NOT_FOUND]?: string[];
            [ComparisonKind.PACKAGE_VERSION_MISMATCH]?: string[];
        };
    };
}

export class Configuration {
    public conf: ConfigurationRecord;

    constructor() {
        // this.conf = {
        //     templates: {
        //         vendor: 'spatie',
        //         names: ['package-skeleton-php', 'package-skeleton-laravel'],
        //     },
        //     templatesPath: `${__dirname}/temp`,
        //     packagesPath: `/development/repositories/spatie`,
        //     scoreRequirements: {
        //         defaults: {
        //             similar: 0.79,
        //             size: 10,
        //         },
        //         files: [
        //             { name: 'CONTRIBUTING.md', scores: { similar: 0.99, size: 5 } },
        //             { name: '.editorconfig', scores: { similar: 1.0, size: 5 } },
        //             { name: '.gitattributes', scores: { similar: 0.9, size: 8 } },
        //             { name: '.gitignore', scores: { similar: 0.9, size: 8 } },
        //             { name: '.php_cs.dist', scores: { similar: 1.0, size: 5 } },
        //         ],
        //     },
        //     skipComparisons: ['composer.json'],
        //     ignoreNames: [
        //         '.git/*',
        //         '.idea/*',
        //         '.idea',
        //         '.vscode/*',
        //         'CHANGELOG.md',
        //         'README.md',
        //         'art/*',
        //         'build/*',
        //         'build',
        //         'composer.lock',
        //         //'config',
        //         'coverage',
        //         'database',
        //         'docs/*',
        //         'node_modules/*',
        //         'package-lock.json',
        //         'resources/*',
        //         'resources',
        //         'src/*',
        //         'src',
        //         'stubs/*',
        //         'tests/*',
        //         'vendor/*',
        //         '*.sh',
        //         '*.cache',
        //         '*.lock',
        //     ],
        // };

        if (process.env.NODE_ENV === 'test') {
            this.conf = this.loadConfigurationFile(process.cwd() + '/tests/data/index.yml').config;
            return;
        }

        this.conf = this.loadConfigurationFile(__filename.replace(/\.[tj]s$/, '.yml')).config;
    }

    public loadConfigurationFile(filename: string) {
        const content = readFileSync(filename, { encoding: 'utf-8' })
            .replace(/\{\{__dirname\}\}/g, __dirname);

        return yaml.load(content);
    }

    public qualifiedTemplateName(templateName: string): string {
        return `${this.conf.templates.vendor}/${templateName}`;
    }

    public qualifiedPackageName(name: string): string {
        return `${this.conf.templates.vendor}/${name}`;
    }

    public getFullTemplateName(shortName: string): string {
        const shortTemplateName = (longName: string) => {
            return longName.split('-')
                .pop() ?? longName;
        };

        return this.conf.templates.names.find(name => shortTemplateName(name) === shortName) ?? shortName;
    }

    public isIssueIgnored(issue: PackageIssue): boolean {
        return this.conf.issues.ignored[issue.result.kind.toString()]?.includes(issue.result.name) ?? false;
    }
}

export const config = new Configuration();
