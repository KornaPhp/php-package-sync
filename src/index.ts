import { loadCommand } from './commands/Command';
import { init } from './lib/PrototypesInit';

init();

const yargs = require('yargs');

console.log(`\n*** READ THE README FIRST TO AVOID UNEXPECTED FILE CHANGES ***`);

yargs(process.argv.slice(2))
    //.scriptName('spatie-package')
    .version('1.0.0')
    .command(loadCommand(require('./commands/Analyze')))
    .command(loadCommand(require('./commands/Fix')))
    .command(loadCommand(require('./commands/PullTemplate')))
    .command(loadCommand(require('./commands/PullPackage')))
    .demandCommand()
    .help()
    .wrap(120).argv;
