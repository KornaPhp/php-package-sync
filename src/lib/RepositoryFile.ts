/* eslint-disable no-unused-vars */

import { ComparisonScoreRequirements } from '../types/ComparisonScoreRequirements';
import { File } from './File';
import { Repository } from './Repository';

export class RepositoryFile extends File {
    public shouldIgnore = false;
    public shouldCompare = false;
    public requiredScores: ComparisonScoreRequirements = { similar: 0, size: 0 };

    constructor(public repository: Repository, protected fn: string, contents: string | null = null) {
        super(fn, contents);
    }

    get relativeName() {
        return this.fn.replace(this.repository.path + '/', '');
    }

    public withOptions(ignore: boolean, compare: boolean, requiredScores: ComparisonScoreRequirements) {
        this.shouldIgnore = ignore;
        this.shouldCompare = compare;
        this.requiredScores = requiredScores;
        //
        return this;
    }
}
