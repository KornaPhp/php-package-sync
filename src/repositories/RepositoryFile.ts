/* eslint-disable no-unused-vars */

import { ComparisonScoreRequirements } from '../types/ComparisonScoreRequirements';
import { File } from '../lib/File';

export class RepositoryFile extends File {
    public shouldIgnore = false;
    public shouldCompare = false;
    public requiredScores: ComparisonScoreRequirements = { similar: 0, size: 0 };

    /**
     *
     * @param repository Repository class
     * @param fn string
     * @param contents string | null
     */
    constructor(public repository: any, protected fn: string, contents: string | null = null) {
        super(fn, contents);
    }

    get relativeName() {
        return this.fn.replace(this.repository.path + '/', '');
    }

    public withOptions(ignore: boolean, compare: boolean, requiredScores: ComparisonScoreRequirements) {
        this.shouldIgnore = ignore;
        this.shouldCompare = compare;
        this.requiredScores = requiredScores;
        //
        return this;
    }

    public processTemplate(): string {
        return super.processTemplate(this.repository.name);
    }
}
