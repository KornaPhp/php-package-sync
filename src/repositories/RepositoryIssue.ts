/* eslint-disable no-unused-vars */

import { ComparisonKind } from '../types/FileComparisonResult';
import { Repository } from './Repository';
import { RepositoryFile } from './RepositoryFile';
//import Fixer from './fixers/Fixer';
import { classOf } from '../lib/helpers';

export class RepositoryIssue {
    protected _fixers: any[] = [];
    public resolvedByFixer = 'none';
    public resolvedNotes: string[] = [];
    public pending = false;

    constructor(
        public result: any,
        public name: string,
        public srcFile: RepositoryFile | null,
        public destFile: RepositoryFile | null,
        public skeleton: Repository,
        public repository: Repository,
        public resolved: boolean = false,
        public note: string | null = null,
        public context: any | null = null,
    ) {
        //
    }

    get fixers() {
        const same = (a, b, method) => (a[method]() && b[method]()) || (!a[method]() && !b[method]());
        const isTrue = (a, method) => (a[method]() ? 1 : -1);
        const methodCompare = (a, b, method) => (same(a, b, method) ? 0 : isTrue(a, method));

        return this._fixers.sort((a, b) => methodCompare(a, b, 'runsFixers'))
            .sort((a, b) => methodCompare(a, b, 'isRisky'));
    }

    get kind(): ComparisonKind {
        return this.result.kind;
    }

    get score(): number | string {
        if (this.result.score === 0) {
            return '-';
        }

        if (typeof this.result.score === 'number') {
            return this.result.score.toFixed(3);
        }

        return this.result.score;
    }

    get sourcefile(): RepositoryFile {
        return <RepositoryFile>this.srcFile;
    }

    get targetfile(): RepositoryFile {
        return <RepositoryFile>this.destFile;
    }

    public addFixer(fixer: any) {
        if (this.fixers.find(f => f.getName() === fixer.getName()) === undefined) {
            this.fixers.push(fixer);
        }

        return this;
    }

    public is(kind: ComparisonKind): boolean {
        return this.kind === kind;
    }

    public resolve(resolvedByFixer: string | any) {
        if (typeof resolvedByFixer !== 'string') {
            resolvedByFixer = classOf(resolvedByFixer)
                .prettyName();
        }

        this.resolvedByFixer = <string>resolvedByFixer;
        this.resolved = true;

        return this;
    }

    public addResolvedNote(note: string, prepend = false) {
        if (prepend) {
            this.resolvedNotes.unshift(note);
            return this;
        }

        this.resolvedNotes.push(note);

        return this;
    }

    public disableFixers(names: string[] | null = null) {
        if (names === null) {
            names = this.fixers.map(fixer => fixer.getName());
        }

        this.fixers.forEach(fixer => {
            if (names?.includes(fixer.getName())) {
                fixer.disable();
            }
        });

        return this;
    }

    public enableFixers(names: string[] | null = null) {
        if (names === null) {
            names = this.fixers.map(fixer => fixer.getName());
        }

        this.fixers.forEach(fixer => {
            if (names?.includes(fixer.getName())) {
                fixer.enable();
            }
        });

        return this;
    }
}
