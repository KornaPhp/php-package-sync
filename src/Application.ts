/* eslint-disable no-unused-vars */

import { existsSync, mkdirSync } from 'fs';
import { basename, sep } from 'path';
import { ComparisonKind } from './types/FileComparisonResult';
import { ComparisonScoreRequirements } from './types/ComparisonScoreRequirements';
import { ComposerComparer } from './lib/composer/ComposerComparer';
import { Configuration } from './Configuration';
import { FileScoreRequirements } from './types/FileScoreRequirements';
import { Repository, RepositoryKind } from './lib/Repository';
import { RepositoryIssue } from './issues/RepositoryIssue';
import { FixerManager } from './issues/FixerManager';
import { Comparisons } from './lib/comparisions/Comparisons';
import { RepositoryFile } from './lib/RepositoryFile';
import { RepositoryValidator } from './lib/RepositoryValidator';

const micromatch = require('micromatch');

export class Application {
    public configuration: Configuration;
    public basePath: string;

    constructor(basePath: string) {
        this.basePath = basePath;
        this.configuration = new Configuration();

        this.ensureStoragePathsExist();
    }

    get config() {
        return this.configuration.conf;
    }

    public ensureStoragePathsExist() {
        if (!existsSync(this.config.templatesPath)) {
            mkdirSync(this.config.templatesPath, { recursive: true });
        }
        if (!existsSync(this.config.packagesPath)) {
            mkdirSync(this.config.packagesPath, { recursive: true });
        }
    }

    public templatePath(templateName: string): string {
        return `${this.config.templatesPath}/${templateName}`;
    }

    public packagePath(packageName: string): string {
        return `${this.config.packagesPath}/${packageName}`;
    }

    public shouldIgnoreFile(fn: string): boolean {
        return (
            micromatch.isMatch(fn, this.config.ignoreNames) || // || micromatch.contains(fn, this.config.ignoreNames);
            micromatch.isMatch(fn.replace(process.cwd() + sep, ''), this.config.ignoreNames)
        );
    }

    public shouldCompareFile(fn: string): boolean {
        return !this.config.skipComparisons.includes(basename(fn));
    }

    public getSimilarScoreRequirement(fn: string): number {
        const reqs = this.config.scoreRequirements;

        return reqs.files.find((req: FileScoreRequirements) => req.name === basename(fn))?.scores?.similar ?? reqs.defaults.similar;
    }

    public getMaxAllowedSizeDifferenceScore(fn: string): number {
        const reqs = this.config.scoreRequirements;

        return reqs.files.find(req => req.name === basename(fn))?.scores?.size ?? reqs.defaults.size;
    }

    public getFileScoreRequirements(fn: string): ComparisonScoreRequirements {
        return {
            similar: this.getSimilarScoreRequirement(fn),
            size: this.getMaxAllowedSizeDifferenceScore(fn),
        };
    }

    performStringComparison(skeleton: Repository, repo: Repository, file: RepositoryFile, repoFile: RepositoryFile | null) {
        const strComparison = Comparisons.strings(file?.processTemplate(), repoFile?.contents);

        if (!strComparison.meetsRequirement(file.requiredScores.similar)) {
            const compareResult = {
                kind: ComparisonKind.FILE_NOT_SIMILAR_ENOUGH,
                score: strComparison.similarityScore,
            };

            repo.issues.push(new RepositoryIssue(compareResult, file.relativeName, file, repoFile, skeleton, repo, false));

            return true;
        }

        return false;
    }

    performSizeComparison(skeleton: Repository, repo: Repository, file: RepositoryFile, repoFile: RepositoryFile | null) {
        const sizeComparison = Comparisons.filesizes(file.sizeOnDisk, repoFile?.sizeOnDisk);

        if (sizeComparison.meetsRequirement(file.requiredScores.size)) {
            const result = {
                kind: ComparisonKind.ALLOWED_SIZE_DIFFERENCE_EXCEEDED,
                score: sizeComparison.format(2),
            };

            repo.issues.push(new RepositoryIssue(result, file.relativeName, file, repoFile, skeleton, repo, false));

            return true;
        }

        return false;
    }

    performFileExistsComparison(skeleton: Repository, repo: Repository, file: RepositoryFile) {
        if (!file.shouldIgnore && !repo.hasFile(file)) {
            const kind = file.isFile() ? ComparisonKind.FILE_NOT_FOUND : ComparisonKind.DIRECTORY_NOT_FOUND;

            repo.issues.push(new RepositoryIssue({ kind, score: 0 }, file.relativeName, file, null, skeleton, repo, false));

            return true;
        }

        return false;
    }

    compareRepositories(skeleton: Repository, repo: Repository) {
        skeleton.files.forEach(file => {
            const repoFile = repo.getFile(file.relativeName);

            if (this.performFileExistsComparison(skeleton, repo, file)) {
                return;
            }

            if (this.performStringComparison(skeleton, repo, file, repoFile)) {
                return;
            }

            if (this.performSizeComparison(skeleton, repo, file, repoFile)) {
                return;
            }
        });

        this.checkRepoForFilesNotInSkeleton(repo, skeleton);

        ComposerComparer.comparePackages(skeleton.path, repo.path)
            .forEach(r =>
                repo.issues.push(new RepositoryIssue(r, r.name, null, null, skeleton, repo, false)),
            );

        ComposerComparer.compareScripts(skeleton.path, repo.path)
            .forEach(r =>
                repo.issues.push(new RepositoryIssue(r, r.name, null, null, skeleton, repo, false)),
            );
    }

    checkRepoForFilesNotInSkeleton(repo: Repository, skeleton: Repository) {
        repo.files
            .filter(file => !file.shouldIgnore)
            .filter(file => !skeleton.hasFile(file))
            .forEach(file => {
                const kind = file.isFile() ? ComparisonKind.FILE_NOT_IN_SKELETON : ComparisonKind.DIRECTORY_NOT_IN_SKELETON;

                repo.issues.push(new RepositoryIssue({ kind, score: 0 }, file.relativeName, null, file, skeleton, repo, false));
            });
    }

    analyzePackage(packageName: string) {
        const skeletonType = packageName.startsWith('laravel-') ? 'laravel' : 'php';
        const templateName = this.configuration.getFullTemplateName(skeletonType);

        const skeletonPath = this.templatePath(templateName);
        const repositoryPath = this.packagePath(packageName);

        RepositoryValidator.ensurePackageExists(packageName);
        RepositoryValidator.ensureTemplateExists(templateName);

        const skeleton = Repository.create(skeletonPath, RepositoryKind.SKELETON);
        const repo = Repository.create(repositoryPath, RepositoryKind.PACKAGE);

        return this.analyzeRepository(skeleton, repo);
    }

    analyzeRepository(skeleton: Repository, repo: Repository) {
        this.compareRepositories(skeleton, repo);

        repo.issues.forEach(issue => {
            FixerManager.fixers()
                .forEach(fixer => {
                    if (fixer.fixes(issue.kind) && fixer.canFix(issue)) {
                        issue.availableFixers.push(fixer.prettyName());
                    }
                });
        });

        return { skeleton, repo };
    }
}

export const app = new Application(process.cwd());
