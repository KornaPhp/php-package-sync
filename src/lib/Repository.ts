/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */

import { basename } from 'path';
import { app } from '../Application';
//import { PackageIssue } from '../issues/PackageIssue';
import { RepositoryIssue } from '../issues/RepositoryIssue';
import { Composer } from './composer/Composer';
import { getFileList, isDirectory } from './helpers';
import { RepositoryFile } from './RepositoryFile';

export enum RepositoryKind {
    SKELETON = 'skeleton',
    PACKAGE = 'package',
}

export class Repository {
    protected composerData: Composer;
    protected loadedFiles = false;
    protected fileList: RepositoryFile[] = [];
    public issues: RepositoryIssue[] = [];

    constructor(public path: string, public kind: RepositoryKind) {
        this.composerData = Composer.createFromPath(path);
    }

    static create(path: string, kind: RepositoryKind) {
        return new Repository(path, kind);
    }

    get name() {
        return basename(this.path);
    }

    get composer() {
        return this.composerData;
    }

    get packages() {
        return this.composer.packages('all');
    }

    get files() {
        if (!this.loadedFiles) {
            this.loadFiles();
        }

        return this.fileList;
    }

    getFile(relativeName: string): RepositoryFile | null {
        return this.files.find(f => f.relativeName === relativeName) || null;
    }

    hasFile(file: RepositoryFile): boolean {
        return this.files.find(f => f.relativeName === file.relativeName) !== undefined;
    }

    hasIssues() {
        return this.issues.length > 0;
    }

    isSkeleton() {
        return this.kind === RepositoryKind.SKELETON;
    }

    isPackage() {
        return this.kind === RepositoryKind.PACKAGE;
    }

    protected getFileList(directory: string, basePath: string | null = null): RepositoryFile[] {
        const result: RepositoryFile[] = [];

        getFileList(directory, basePath, true)
            .filter(fn => !app.shouldIgnoreFile(fn.replace(`${basePath}/`, '')))
            .forEach(fqName => {
                const rfile = new RepositoryFile(this, fqName);

                const relativeName = basePath ? fqName.replace(`${basePath}/`, '') : fqName;
                const isFile = !isDirectory(fqName);

                if (!result.find(item => item.relativeName === relativeName)) {
                    result.push(
                        rfile.withOptions(
                            app.shouldIgnoreFile(relativeName),
                            rfile.isFile() && app.shouldCompareFile(fqName),
                            app.getFileScoreRequirements(fqName),
                        ),
                    );
                    //     {
                    //     isFile: !isPath,
                    //     name: fqName,
                    //     relativeName: relativeName,
                    //     filesize: fileSize(fqName),
                    //     shouldIgnore: app.shouldIgnoreFile(relativeName),
                    //     shouldCompare: !isPath && app.shouldCompareFile(fqName),
                    //     requiredScores: app.getFileScoreRequirements(fqName),
                    // });

                    if (rfile.isDirectory()) {
                        result.push(...this.getFileList(fqName, basePath ?? directory));
                    }
                }
            });

        return result;
    }

    public loadFiles() {
        this.fileList = this.getFileList(this.path, this.path);
        this.loadedFiles = true;

        return this;
    }
}
