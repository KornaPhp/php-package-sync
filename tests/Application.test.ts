/* eslint-disable no-undef */

import { Application } from '../src/Application';
import { Configuration } from '../src/Configuration';
import { Repository, RepositoryKind } from '../src/repositories/Repository';
import { RepositoryFile } from '../src/repositories/RepositoryFile';
import { ComparisonKind } from '../src/types/FileComparisonResult';

let conf: Configuration, app: Application, skeleton: Repository, repo: Repository;

beforeEach(() => {
    conf = new Configuration(__dirname + '/../data/index.yml');
    app = new Application(conf);

    skeleton = Repository.create(__dirname + '/data/test-skeleton', RepositoryKind.SKELETON);
    repo = Repository.create(__dirname + '/data/test-package-2', RepositoryKind.PACKAGE);

    skeleton.loadFiles();
    repo.loadFiles();
});

it('performs a file exists comparison', () => {
    const file = <RepositoryFile>skeleton.getFile('testfile1.txt');
    const result = app.performFileExistsComparison(skeleton, repo, file);

    expect(result).toBeTruthy();
    expect(repo.issues).toHaveLength(1);
    expect(repo.issues[0].kind).toBe(ComparisonKind.FILE_NOT_FOUND);
});

it('performs a dir exists comparison', () => {
    const file = <RepositoryFile>skeleton.getFile('SOME_DIR');
    const result = app.performFileExistsComparison(skeleton, repo, file);

    expect(result).toBeTruthy();
    expect(repo.issues).toHaveLength(1);
    expect(repo.issues[0].kind).toBe(ComparisonKind.DIRECTORY_NOT_FOUND);
});
