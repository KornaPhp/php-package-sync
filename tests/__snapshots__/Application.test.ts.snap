// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the repo for files not in the template 1`] = `
Array [
  Object {
    "name": "one.txt",
    "resolvedNotes": Array [],
    "result": Object {
      "kind": "extra_file",
      "score": 0,
    },
    "score": "-",
  },
  Object {
    "name": "two.txt",
    "resolvedNotes": Array [],
    "result": Object {
      "kind": "extra_file",
      "score": 0,
    },
    "score": "-",
  },
]
`;
